homeassistant/components/stream/hls.py

```diff
@@ -83,15 +84,31 @@ def render_playlist(track):
         if not segments:
             return []
 
+        first_segment = segments[0]
         playlist = [
-            f"#EXT-X-MEDIA-SEQUENCE:{segments[0].sequence}",
-            f"#EXT-X-DISCONTINUITY-SEQUENCE:{segments[0].stream_id}",
+            f"#EXT-X-MEDIA-SEQUENCE:{first_segment.sequence}",
+            f"#EXT-X-DISCONTINUITY-SEQUENCE:{first_segment.stream_id}",
+            "#EXT-X-PROGRAM-DATE-TIME:"
+            + first_segment.start_time.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3]
+            + "Z",
+            # Since our window doesn't have many segments, we don't want to start
+            # at the beginning or we risk a behind live window exception in exoplayer.
+            # EXT-X-START is not supposed to be within 3 target durations of the end,
+            # but this seems ok
+            f"#EXT-X-START:TIME-OFFSET=-{1.5 * track.target_duration:.3f},PRECISE=YES",
```

Comment:
If i understand this right, it may start the live player closer to realtime than we today, which is at the oldest segment in the buffer.  Is that right? This seems like a really nice performance trade off that seems right in the context of home assistant where people would probably rather have closer to real time even if it risks some buffering of catching up to the front of the stream.  If i understand that right, maybe change the title like: `Reduce perceived video stream latency` or something?  Super cool.

I remember being on discord with @hunterjm and he showed me that when the video player starts you can effectively advance it a few seconds yourself -- and this effectively does that for you.

Reply:
Exactly. Just need to make sure it works on iOS